bin/ddg2logo

```diff
@@ -281,11 +289,23 @@ for fname in fnames:
 
 data = np.vstack(tuple(data))
 
-# set values above the maximum value to the maximum
-# move the whole range of values, so that values at maximum become 0
-# and invert sign so that higher values are more stabilizing
-data[ data >= options.threshold ] = options.threshold
-data = (data - options.threshold) * (-1)
+# set values above the threshold (for stabilizing mutations) or below
+# it (for the destabilizing ones) to the threshold value
+# move the whole range of values, so that values at maximum
+# (stabilizing) or at minimum (destabilizing)  become 0
+# In the case of stabilizing mutations, invert sign so that higher values are more stabilizing
+if options.threshold:
+    try:
+        data[ data >= options.threshold ] = options.threshold
+        data = (data - options.threshold) * (-1)
+    except:
+        exit(1)
```

Comment:
what cases is this exception supposed to cover?

exiting in case of exception here isn't very good as 1) no output is produced 2) no error message is produced detailing what the error was. The script just exits without any feedback which can be very confusing. If you want to use exceptions here you need to be specific on what you're catching and report feedback appropriately

Reply:
Since I wasn't thinking to any exception, I just removed the try-exception code. Also because the only possible error is that you don't specify at least one argument, but that is already covered .